extensions-core/orc-extensions/src/main/java/org/apache/druid/data/input/orc/OrcStructConverter.java

```diff
@@ -0,0 +1,198 @@
+      OrcMap<? extends WritableComparable, ? extends WritableComparable> map
+  )
+  {
+    Map<Object, Object> converted = new HashMap<>();
+    TypeDescription keyDescription = fieldDescription.getChildren().get(0);
+    TypeDescription valueDescription = fieldDescription.getChildren().get(1);
+    for (WritableComparable key : map.navigableKeySet()) {
+      Object newKey = convertPrimitive(keyDescription, key);
+      if (valueDescription.getCategory().isPrimitive()) {
+        converted.put(newKey, convertPrimitive(valueDescription, map.get(key)));
+      } else {
+        converted.put(newKey, map.get(key));
+      }
+    }
+    return converted;
+  }
+
+  private Map<String, Object> convertMap(OrcStruct map)
+  {
+    Map<String, Object> converted = new HashMap<>();
+    for (String key : map.getSchema().getFieldNames()) {
+      converted.put(key, convertField(map, key));
+    }
+    return converted;
+  }
+
+  @SuppressForbidden(reason = "new DateTime(java.util.Date)")
+  private static Object convertPrimitive(TypeDescription fieldDescription, WritableComparable field)
+  {
+    /*
+        ORC TYPE    WRITABLE TYPE
+        binary      org.apache.hadoop.io.BytesWritable
+        bigint      org.apache.hadoop.io.LongWritable
+        boolean     org.apache.hadoop.io.BooleanWritable
+        char        org.apache.hadoop.io.Text
+        date        org.apache.hadoop.hive.serde2.io.DateWritable
+        decimal     org.apache.hadoop.hive.serde2.io.HiveDecimalWritable
+        double      org.apache.hadoop.io.DoubleWritable
+        float       org.apache.hadoop.io.FloatWritable
+        int         org.apache.hadoop.io.IntWritable
+        smallint    org.apache.hadoop.io.ShortWritable
+        string      org.apache.hadoop.io.Text
+        timestamp   org.apache.orc.mapred.OrcTimestamp
+        tinyint     org.apache.hadoop.io.ByteWritable
+        varchar     org.apache.hadoop.io.Text
+     */
+    switch (fieldDescription.getCategory()) {
+      case STRING:
+      case CHAR:
+      case VARCHAR:
+        return ((Text) field).toString();
+      case BOOLEAN:
+        return ((BooleanWritable) field).get();
+      case BYTE:
+        return ((ByteWritable) field).get();
+      case SHORT:
+        return ((ShortWritable) field).get();
+      case INT:
+        return ((IntWritable) field).get();
+      case LONG:
+        return ((LongWritable) field).get();
+      case FLOAT:
+        return ((FloatWritable) field).get();
+      case DOUBLE:
+        return ((DoubleWritable) field).get();
+      case DECIMAL:
+        return ((HiveDecimalWritable) field).getHiveDecimal().doubleValue();
+      case TIMESTAMP:
+        return ((OrcTimestamp) field).getTime();
+      case DATE:
+        // todo: is the the best way to go from java.util.Date to DateTime?
+        return new DateTime(((DateWritable) field).get());
+      case BINARY:
+        // todo: hmm, i think we need a standard way of handling binary blobs this is a placeholder.
```

Comment:
Please don't commit TODOs, either implement it or make it a less todo-y comment, like `// Base64-encode binary blobs; may want to provide other options in the future.`

Reply:
Oops, I left the todo because I considered this PR still sort of wip pending feedback on the proposal, and wanted to have a discussion about what to do with binary.

Avro handles binary by optionally converting to utf8 string or just returning a byte array:
```
...
  if (field instanceof ByteBuffer) {
      if (binaryAsString) {
        return StringUtils.fromUtf8(((ByteBuffer) field).array());
      } else {
        return ((ByteBuffer) field).array();
      }
    }
....
```

With the parquet extension, I decided to mimic this behavior without thinking very hard about it, because I wanted to maintain compatibility with the avro conversion since that's what the extension was previously using, so it also does:
```
...
          case FIXED_LEN_BYTE_ARRAY:
          case BINARY:
            Binary bin = g.getBinary(fieldIndex, index);
            byte[] bytes = bin.getBytes();
            if (binaryAsString) {
              return StringUtils.fromUtf8(bytes);
            } else {
              return bytes;
            }
...
```

The previous version of this extension looks like it had no special handling for binary fields, so would eventually just tostring on the byte array.

My thoughts are that we should maybe preserve the `binaryAsString` functionality of avro and parquet, and also apply it here, and make this extension also return byte[] if it shouldn't treat it as a utf8 string rather than just always translating to base64 encoded string like is being done here currently. I do think that we should consider converting these byte[] objects into base64 strings down the line if they end up in a string dimension, I'm not certain where that should happen though.

Reply:
It looks to me like the correct place to handle conversion of `byte[]` to a base64 string if it ends up as a string dimension would be `Rows.objectToStrings`. I think conversion to base64 would at least make the dimension more usable than the result of calling `byte[].toString` which is something like `[B@73809e7`.

Reply:
I have made this change to `Rows.objectToStrings`, to special handle `byte[]` and convert to base64.